crates/core/src/lib.rs

```diff
@@ -47,13 +63,57 @@ impl Config {
     pub fn wasmtime_config(&mut self) -> &mut wasmtime::Config {
         &mut self.inner
     }
+
+    /// Enable the Wasmtime compilation cache with the given path, if any, to load configuration from.
+    pub fn configure_cache(&mut self, config_path: &Option<PathBuf>) -> Result<()> {
+        match config_path {
+            Some(p) => self.inner.cache_config_load(p)?,
+            None => self.inner.cache_config_load_default()?,
+        };
+
+        Ok(())
+    }
+
+    /// Enable or update parameters for the pooling instance allocator.
+    #[cfg(not(target_os = "windows"))]
+    pub fn enable_pooling(
+        &mut self,
+        max_memories: u32,
+        max_memory_pages: u64,
+        max_tables: u32,
+        max_table_entries: u32,
```

Comment:
I wonder if we can do a better job of selecting reasonable defaults here. The implications of these settings aren't going to be obvious to most people.